src/2015-04-25-epidemiological-compartment-model.cpp

```diff
@@ -0,0 +1,240 @@
+/**
+ * @title Stochastic SIR Epidemiological Compartment Model
+ * @author Christian Gunning
+ * @license GPL (>= 2)
+ * @tags simulation parameter
+ * @summary Demonstrates a discrete, stochastic epidemiological 
+ *  using a tau-leap method.  This model 
+ *   takes a list of parameters and returns a data.frame
+ *   of simulation results.
+ *
+ * ### Introduction
+ *
+ * This post is a simple introduction to Rcpp for disease
+ * ecologists, epidemiologists, or dynamical systems 
+ * modelers - the sorts of folks who will benefit from a 
+ * simple but fully-working example.
+ * My intent is to provide a complete, 
+ * self-contained introduction to modeling with Rcpp.  
+ * My hope is that this model can be easily modified 
+ * to run any dynamical simulation that has 
+ * dependence on the previous time step 
+ * (and can therefore not be vectorized).
+ *
+ * This post uses a classic 
+ * Susceptible-Infected-Recovered (SIR)
+ * [epidemiological compartment model](https://en.wikipedia.org/wiki/Compartmental_models_in_epidemiology).
+ * Compartment models are simple, commonly-used dynamical
+ * systems models.  Here I demonstrate the
+ * [tau-leap method](https://en.wikipedia.org/wiki/Tau-leaping), where
+ * a discrete number of individuals move probabilistically 
+ * between compartments at fixed intervals in time.
+ * In this model, the wait-times within class 
+ * are exponentially distributed, and the number of
+ * transitions between states in a fixed time step are 
+ * Poisson distributed.
+ *
+ * This model is parameterized for the spread of measles
+ * in a closed population, 
+ * where the birth rate (nu) = death rate (mu).
+ * The transmission rate (beta) describes how frequently 
+ * susceptible (S) and infected (I) individuals come into 
+ * contact, and the recovery rate (gamma) describes the
+ * the average time an individual spends infected before
+ * recovering. 
+*/
+
+/** 
+ * ### C++ Code
+ * Note that cpp Functions must be marked with the following comment like for use in R
+ * // [[Rcpp::export]]
+*/
+
+#include <Rcpp.h>
+// Declare commonly used objects
+using Rcpp::NumericVector;
+using Rcpp::List;
+using Rcpp::DataFrame;
+using Rcpp::Named;
+// use Rcpp namespace R:: (alias to R API)
+// R::rpois always returns a single value
+// to return multiple (e.g. Integer/NumericVector, 
+// use Rcpp::rpois(int ndraw, param) and friends
+using R::rpois;
```

Comment:
Here I'd prefer we not employ `using` at all (so that it's clear at the call site that we are calling the R api rather than a locally defined function or Rcpp API).